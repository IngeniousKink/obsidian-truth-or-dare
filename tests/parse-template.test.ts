import { fromMarkdown } from 'mdast-util-from-markdown';
import { Paragraph } from 'node_modules/mdast-util-from-markdown/lib/index.js';

describe('parse-template', () => {

  test('document how mast-util-from-markdown parses images', () => {

    const fileContents = '123![](https://example.com/favicon.ico)456';

    const root = fromMarkdown(fileContents);
    
    const para = root.children[0] as Paragraph;

    expect(para.children).toMatchSnapshot();
    
    expect(para.children.length).toBe(3);
    expect(para.children[0].type).toBe('text');
    expect(para.children[1].type).toBe('image');
    expect(para.children[2].type).toBe('text');

  });


});



