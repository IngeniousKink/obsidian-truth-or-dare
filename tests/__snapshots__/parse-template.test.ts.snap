// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse-template > document how mast-util-from-markdown parses images 1`] = `
[
  {
    "position": {
      "end": {
        "column": 4,
        "line": 1,
        "offset": 3,
      },
      "start": {
        "column": 1,
        "line": 1,
        "offset": 0,
      },
    },
    "type": "text",
    "value": "123",
  },
  {
    "alt": "",
    "position": {
      "end": {
        "column": 40,
        "line": 1,
        "offset": 39,
      },
      "start": {
        "column": 4,
        "line": 1,
        "offset": 3,
      },
    },
    "title": null,
    "type": "image",
    "url": "https://example.com/favicon.ico",
  },
  {
    "position": {
      "end": {
        "column": 43,
        "line": 1,
        "offset": 42,
      },
      "start": {
        "column": 40,
        "line": 1,
        "offset": 39,
      },
    },
    "type": "text",
    "value": "456",
  },
]
`;
